pkgs/servers/http/nginx/generic.nix

```diff
@@ -94,6 +93,8 @@ stdenv.mkDerivation {
     "--with-perl_modules_path=lib/perl5"
   ]
     ++ optional (gd != null) "--with-http_image_filter_module"
+    ++ optional (geoip != null) "--with-http_geoip_module"
+    ++ optional (withStream && geoip != null) "--with-stream_geoip_module"
```

Comment:
Please replace this with a boolean flag like enableGeoIP.

Reply:
Why? The gd dependency doesn't have a boolean flag either. I'd rather keep it consistent.

Reply:
It can be removed.
It is geoip version 1, which is no longer supported upstream (no fresh databases since many months ago).
No one will need `enableGeoIP=true`

Reply:
`pkgs.geoip` can be removed too

Reply:
I don't entirely agree with that. While there are no official updates to the legacy GeoIP databases, there are third party providers for recent data in the legacy GeoIP format, which is converted from GeoIP2 data.
Some applications (like nginx, at least officially) don't have support for the new GeoIP2 format yet.
So I think we should keep geoip around for another year until it's support period runs out, and try to add a module that regularly downloads current GeoIP legacy data from https://mailfud.org/geoip-legacy/.

Reply:
> Why? The gd dependency doesn't have a boolean flag either. I'd rather keep it consistent.

I might argue that this dependency should also have a boolean flag if it is something someone would conceivably want to turn off...

Reply:
(asking to switch the conventions in an unrelated patch makes sense for coding style rules written down in «Coding conventions» in the manual, and null-overriding will go nowhere without an actual written down decision)